Index: lxsession-0.4.9.2+git20140410/configure.ac
===================================================================
--- lxsession-0.4.9.2+git20140410.orig/configure.ac	2014-06-06 21:19:30.656338182 +0200
+++ lxsession-0.4.9.2+git20140410/configure.ac	2014-06-06 21:19:30.652338284 +0200
@@ -1,7 +1,7 @@
 AC_INIT([lxsession],[0.4.9.3],[],[lxsession])
 
 # Autotols programs
-AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror foreign no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([1.11.1 -Wall foreign no-dist-gzip dist-xz])
 AC_PREREQ([2.68])
 IT_PROG_INTLTOOL([0.40.0],[no-xml])
 
